hathor/indexes/rocksdb_tokens_index.py

```diff
@@ -0,0 +1,463 @@
+        import struct
+        assert len(key) >= 1
+        tag = _Tag(key[0])
+        token_uid = bytes(key[1:33])
+        assert len(token_uid) == 32
+        if tag is _Tag.INFO:
+            assert len(key) == 32 + 1
+            return _KeyAny(token_uid, tag)
+        elif tag is _Tag.TXS:
+            assert len(key) == 32 + 1 + 4 + 32
+            timestamp: int
+            (timestamp,) = struct.unpack('>I', key[33:37])
+            tx_hash = key[37:]
+            assert len(tx_hash) == 32
+            return _KeyAny(token_uid, tag, timestamp=timestamp, tx_hash=tx_hash)
+        elif tag in {_Tag.MINT, _Tag.MELT}:
+            assert len(key) == 32 + 1 + 32 + 1
+            tx_hash = key[33:65]
+            index = key[65]
+            return _KeyAny(token_uid, tag, tx_hash=tx_hash, index=index)
+        else:
+            # XXX: if/elif is exhaustive for all possible tags and invalid tag value will fail sooner
+            raise NotImplementedError('unreachable')
+
+    def _to_value_info(self, info: _InfoDict) -> bytes:
+        return json_dumpb(info)
+
+    def _from_value_info(self, value: bytes) -> _InfoDict:
+        return cast(_InfoDict, json_loadb(value))
+
+    def _create_token_info(self, token_uid: bytes, name: str, symbol: str, total: int = 0) -> None:
+        key = self._to_key_info(token_uid)
+        old_value = self._db.get((self._cf, key))
+        assert old_value is None
+        value = self._to_value_info({
+            'name': name,
+            'symbol': symbol,
+            'total': total,
+        })
+        self._db.put((self._cf, key), value)
+
+    def _destroy_token(self, token_uid: bytes) -> None:
+        import rocksdb
+
+        # XXX: this can be a large operation, a writebatch works similar to a "SQL transaction" in that if there is a
+        #      failure it'll either persist all operations or none
+        key_info = self._to_key_info(token_uid)
+        key_mint = self._to_key_mint(token_uid, None)
+        key_melt = self._to_key_mint(token_uid, None)
+        key_txs = self._to_key_txs(token_uid, None)
+        batch = rocksdb.WriteBatch()
+        batch.delete((self._cf, key_info))
+        it = self._db.iterkeys(self._cf)
+        for seek_key in [key_mint, key_melt, key_txs]:
+            it.seek(seek_key)
+            for _, key in it:
+                if not key.startswith(key_mint):
+                    break
+                batch.delete((self._cf, key))
+        self._db.write(batch)
+
+    def _add_transaction(self, token_uid: bytes, timestamp: int, tx_hash: bytes) -> None:
+        self._db.put((self._cf, self._to_key_txs(token_uid, _TxIndex(tx_hash, timestamp))), b'')
+
+    def _remove_transaction(self, token_uid: bytes, timestamp: int, tx_hash: bytes) -> None:
+        self._db.delete((self._cf, self._to_key_txs(token_uid, _TxIndex(tx_hash, timestamp))))
+
+    def _add_mint_utxo(self, token_uid: bytes, txid: bytes, index: int) -> None:
```

Comment:
We've been using `tx_hash` in the whole file, maybe we could change this `txid` to `tx_hash` just for consistency, what do you think?

I'm okay to move with this as well.

Reply:
I'm OK either way. I went with `txid` because it's what we use when talking about inputs, but I just realized we actually use `tx_id` in code.

Reply:
Renamed to `tx_hash`.